harvard-university-for-the-creative-arts.csl

```diff
@@ -167,20 +203,25 @@
           </group>
         </group>
       </else-if>
-      <else-if type="map" match="any">
-        <group delimiter=" ">
-          <text variable="genre" prefix="[" suffix="]"/>
-          <text variable="scale" suffix="."/>
-          <text variable="publisher-place" suffix=":"/>
-          <text variable="publisher" suffix="."/>
-          <text variable="collection-title" prefix="(" suffix=")"/>
-        </group>
-      </else-if>
       <else>
-        <group delimiter=": ">
-          <text variable="publisher-place"/>
-          <text variable="publisher" suffix="."/>
-        </group>
+        <choose>
+          <if match="any" variable="publisher" type="motion_picture broadcast report musical_score song entry-dictionary entry-encyclopedia">
+            <text variable="publisher" suffix="."/>
+          </if>
+          <else>
+            <text value="s.n." prefix="(" suffix=")."/>
+          </else>
+        </choose>
+      </else>
+    </choose>
+  </macro>
+  <macro name="publisher-place">
+    <choose>
+      <if match="any" variable="publisher-place" type="broadcast motion_picture report musical_score song entry-dictionary entry-encyclopedia manuscript">
+        <text variable="publisher-place"/>
+      </if>
```

Comment:
So, you want the publisher-place variable printed, with a fallback to "s.l." unless the item, a) is of the type thesis, manuscript, broadcast, article-newspaper, or b) has a non-empty URL variable. I haven't tested it, but I think you could achieve that with the more compact:

```xml
  <macro name="publisher-place">
    <choose>
      <if variable="publisher-place">
        <text variable="publisher-place"/>
      </if>
      <else-if type="thesis manuscript broadcast article-newspaper" variable="URL" match="none">
        <text value="s.l." prefix="(" suffix=")"/>
      </else-if>
    </choose>
  </macro>
```